variables.tf

```diff
@@ -61,18 +61,16 @@ variable "skip_kms_iam_authorization_policy" {
   default     = false
 }
 
-variable "existing_kms_instance_guid" {
-  type        = string
-  description = "The GUID of the Hyper Protect Crypto Services or Key Protect instance in which the key specified in `kms_key_crn` is coming from. Required only if `kms_encryption_enabled` is set to true, and `skip_kms_iam_authorization_policy` is set to false."
-  default     = null
-}
-
 variable "kms_key_crn" {
   type        = string
   description = "The root key CRN of a Key Management Service like Key Protect or Hyper Protect Crypto Services (HPCS) that you want to use for encryption. Only used if `kms_encryption_enabled` is set to true."
   default     = null
 }
 
+variable "is_hpcs_key" {
+  type        = bool
+  description = "Set it to true if the key is Hyper Protect Crypto Services key"
```

Comment:
I would default this to false as most consumers probably use Key Protect. The variable description should reference the `kms_key_crn` inpout